File: backend/host/Models/Restaurant.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using host.DataBaseAccess;
using host.Logic;
using Microsoft.EntityFrameworkCore;

namespace host.Models;

public class Restaurant : ITableDataBase, ITakeRelatedData
{
    public int? Id { get; set; }
    
    [JsonPropertyName("owner_id")]
    public int OwnerId { get; set; }
    public string Name { get; set; }
    public GeoCoordinates Coordinates { get; set; }
    
    [JsonPropertyName("notification_getter")]
    [NotMapped] public NotificationGetter? NotificationGetter { get; set; }
    [NotMapped] public Menu? Menu { get; set; }
    [NotMapped] public List<Table>? Tables { get; set; }
    
    public Restaurant(){}
    public Restaurant(int? id, int ownerId, string name, GeoCoordinates coordinates, NotificationGetter? notificationGetter,
        Menu? menu, List<Table>? tables)
    {
        Id = id;
        OwnerId = ownerId;
        Name = name;
        Coordinates = coordinates;
        NotificationGetter = notificationGetter;
        Menu = menu;
        Tables = tables;
    }

    public async Task TakeRelatedData(ApplicationDbContext context)
    {
        Menu = await context.Menus
            .FirstOrDefaultAsync(m => m.RestaurantId == Id);

        if (Menu == null)
            throw new Exception("Menu is null");
        

        if (Menu is ITakeRelatedData relatedDataLoaderMenu)
            await relatedDataLoaderMenu.TakeRelatedData(context);
        
        NotificationGetter = await context.NotificationGetters.FirstOrDefaultAsync(n => n.RestaurantId == Id);

        Tables = await context.Tables
            .Where(t => t.RestaurantId == Id)
            .ToListAsync();
        
    }
}

File: frontend/Business/tg_business_bot/Models/Restaurant.py
from pydantic import BaseModel
from typing import List, Optional

from tg_business_bot.Models.NotificationGetter import NotificationGetter
from tg_business_bot.Models.Menu import Menu
from tg_business_bot.Models.Table import Table


class Restaurant(BaseModel):
    id: Optional[int]
    owner_id: int
    notification_getter: Optional[NotificationGetter]
    name: str
    address: str
    menu: Optional[Menu]
    tables: Optional[List[Table]]

    def __init__(self,
                 owner_id: int,
                 name: str,
                 address: str,
                 id: Optional[int] = None,
                 notification_getter: Optional[NotificationGetter] = None,
                 menu: Optional[Menu] = None,
                 tables: Optional[List[Table]] = None):
        super().__init__(
            id=id,
            owner_id=owner_id,
            notification_getter=notification_getter,
            name=name,
            address=address,
            menu=menu,
            tables=tables
        )
